core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/Flow2Handler.java

```diff
@@ -136,7 +143,16 @@ public void accept(Event<? extends Payload> event) {
                     LOGGER.debug("flow control event arrived: key: {}, flowid: {}, payload: {}", key, flowId, payload);
                     Flow flow = runningFlows.get(flowId);
                     if (flow != null) {
-                        flowLogService.save(flowId, flowChainId, key, payload, flow.getVariables(), flow.getFlowConfigClass(), flow.getCurrentState());
+                        try {
+                            transactionService.required(() -> {
+                                flowLogService.updateLastFlowLogStatus(flow.getFlowId(), failHandledEvents.contains(key));
+                                flowLogService.save(flow.getFlowId(), flowChainId, key, payload, flow.getVariables(),
+                                        flow.getFlowConfigClass(), flow.getCurrentState());
+                                return null;
+                            });
+                        } catch (TransactionService.TransactionExecutionException e) {
+                            LOGGER.error("Failed update last flow log status and save new flow log entry.");
+                        }
                         flow.sendEvent(key, payload);
```

Comment:
In previous implementation if something went wrong during persisting flowlog, the function failed and didn't send the event. In current, the send happens always. My suggestion is to keep original behavior, on case of any error don't change the state of the flow. Plus would be nice to remove it from `runningFLows` to avoid memory leak. Thera are 3 options to restart it, retry, CB restart, other CB in the cluster should pick it up.

Reply:
I agree.